ntp-udp/src/lib.rs

```diff
@@ -48,13 +76,43 @@ impl UdpSocket {
         loop {
             let mut guard = self.io.readable().await?;
             match guard.try_io(|inner| recv(inner.get_ref(), buf)) {
-                Ok(result) => return result,
-                Err(_would_block) => continue,
+                Ok(result) => {
```

Comment:
This would be easier to follow if you did an early continue for the `Err` case, reducing rightward drift for all the code in the `Ok` case.